src/server.c

```diff
@@ -5724,11 +5710,12 @@ sds genRedisInfoString(dict *section_dict, int all_sections, int everything) {
 
     /* Stats */
     if (all_sections  || (dictFind(section_dict,"stats") != NULL)) {
+        pause_event *pause_oom_ev = &(server.client_pause_per_purpose[PAUSE_OOM_THROTTLE]);
         long long stat_total_reads_processed, stat_total_writes_processed;
         long long stat_net_input_bytes, stat_net_output_bytes;
         long long stat_net_repl_input_bytes, stat_net_repl_output_bytes;
-        long long current_client_paused_oom_time = server.stat_current_client_paused_oom_time ?
-             (long long) elapsedUs(server.stat_current_client_paused_oom_time): 0;    
+        long long current_client_paused_oom_time = (pause_oom_ev->end < server.mstime) ?
+            0 : server.mstime - pause_oom_ev->start;
```

Comment:
did we switch from monotonic clock to wall clock?

Reply:
Since this value is being used  only for statistics, i prefered inaccuracy and save execution time.  
I feel uncomfortable to "pollute" top level of `struct redisServer`. I think we should think how to break it down. 
I will rollback my commit. 

I assume when you say "separating the if that decides which action to take,  from the if..." you mean follow the pattern identify the state and "goto update_metrics;"

Reply:
yes, let's first decide on the action to take (start or stop pause).
and then, separately update the statistics, this is what we do in the other similar stats, and AFAICT it doesn't break on RESETSTAT.